src/ucp/core/ucp_context.c

```diff
@@ -1222,6 +1227,32 @@ static ucs_status_t ucp_fill_config(ucp_context_h context,
         }
     }
 
+    /* Parse the sockadrr transports priority list */
+    context->config.num_sockaddr_tls = config->sockaddr_cm_tls.count;
+
+    /* Make sure that a list of sockadrr transports exists and save it */
+    if (config->sockaddr_cm_tls.count == 0) {
+        ucs_warn("No sockaddr transports specified");
+        goto out;
+    }
+
+    num_sockaddr_tls = config->sockaddr_cm_tls.count;
+    /* Allocate an array to hold the sockaddr transports list */
+    context->config.sockadrr_tls = ucs_calloc(num_sockaddr_tls,
```

Comment:
save an array of resource indexes on the context instead of names (tl_id array)
instead of searching for them during client/server flow